Moonscraper Chart Editor/Assets/Scripts/Game/Charts/IO/Midi/MidReader.cs

```diff
@@ -920,18 +853,19 @@ static uint CalculateSustainLength(Song song, NoteOnEvent noteEvent)
         {
             uint tick = (uint)noteEvent.AbsoluteTime;
             var sus = (uint)(noteEvent.OffEvent.AbsoluteTime - tick);
-            int rbSustainFixLength = (int)(64 * song.resolution / SongConfig.STANDARD_BEAT_RESOLUTION);
-            if (sus <= rbSustainFixLength)
+            int susCutoff = (int)(song.resolution / 3); // 1/12th note
+            if (sus <= susCutoff)
```

Comment:
Better to add a value to SongConfig called something like "const int TWELTH_NOTE = STANDARD_BEAT_RESOLUTION / 64", otherwise this code is no longer dependent on the config.

Reply:
Is there anything that not doing so would break? My reasoning behind this was that it should result in exactly the same thing, and it's much clearer this way that that is indeed just cutting off notes less than a 1/12th note in the chart.

The previous formula is starting from a 1/12th note in 192 res, multiplying it by the resolution of the chart, and then dividing it by 192 to get a 1/12th note in the chart's resolution, when you can get the same thing by just dividing the resolution directly by 3. If the song config were to change this would break regardless, since the magic number of 64 in that formula isn't dependent on the song config either.

Reply:
Upon coming across the HOPO threshold calculation, I realize doing it this way doesn't allow customizing the exact threshold if desired. I'll just go ahead and make that 64 a constant.